### ISSUE TITLE
Returning exceptions

### ISSUE THREAD

**Issue by michaelsteigman** [zulip/fakeldap#23]
Has consideration been given to allowing tests to pass in an exception to return? E.g. 

self.mock_ldap.set_return_value(
    "simple_bind_s", ("cn=admin,dc=mgb,dc=org", "ldaptest"), ldap.SERVER_DOWN
)`

**Comment by timabbott** [comment]
I haven't thought about this (though maybe the original maintainer did -- we have mostly done bug fixes and Python version support updates since adopting the project).  I would try to follow the patterns of the Python `mock` library for this sort of thing; and so I wouldn't conflate a return value (returning an exception object) with raising an exception in the syntax.  See https://docs.python.org/3/library/unittest.mock.html ; I think we probably would want to call this `side_effect`.

Patches that come with tests are welcome!

**Comment by michaelsteigman** [comment]
Yes, of course. I’ve used that pattern for this type of testing in the past.

Ok, if time allows, I’ll put something together and submit a PR. Thanks!
